airflow/decorators/base.py

```diff
@@ -480,6 +480,8 @@ def _expand(self, expand_input: ExpandInput, *, strict: bool) -> XComArg:
         except AttributeError:
             operator_name = self.operator_class.__name__
 
+        task_display_name = partial_kwargs.get("task_display_name", None)
```

Comment:
We don’t need to pop the value out at all. It is easier to just add a property on MappedOperator that reads the field instead. (The current implementation is actually wrong since the same is not done in the non-subclassed MappedOperator.)